lib/diagnostic.rb

```diff
@@ -13,28 +19,35 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.reduce(0) { |m, p| m += p }
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the numbers enumerable are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.each do |num|
+      if num.odd?
+        return true
+      end
+    end
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # numbers enumerable are even
   def every_even(numbers)
-    # Your code goes here
+    sum = numbers.reduce(0) { |memo, obj| memo += obj }
+    sum % 2 == 0
```

Comment:
This works, but the solution is a bit 'clever' - it's not super applicable in different circumstances (e.g. if we'd asked for them all to be odd, or square).

Enumberable's `.all?` method, in contrast, is very versatile.
Consider the following cases.
`numbers.all?(&:even?)` 
`numbers.all?(&:odd?)`
```
def is_square? (num)
    ... 
end
numbers.all? { |num| is_square?(num) }
```